### PR TITLE
simplify pipeable, unroll bind_back_fn

### PR DESCRIPTION
**Pull request by ericniebler**

### CODE REVIEW

File: include/meta/meta_fwd.hpp [hunk 1]
```diff
@@ -174,6 +174,24 @@
 #define META_IS_SAME(...) std::is_same<__VA_ARGS__>::value
 #endif
 
+#if defined(__GNUC__)
+#define META_IS_BASE_OF(...) __is_base_of(__VA_ARGS__)
```
Comment (CaseyCarter) [inline include/meta/meta_fwd.hpp]: Clang and MSVC also support this intrinsic with the same name.

File: include/meta/meta_fwd.hpp [hunk 2]
```diff
@@ -174,6 +174,24 @@
...
+#elif META_CXX_TRAIT_VARIABLE_TEMPLATES
+#define META_IS_BASE_OF(...) std::is_base_of_v<__VA_ARGS__>
+#else
+#define META_IS_BASE_OF(...) std::is_base_of<__VA_ARGS__>::value
+#endif
+
+#if defined(__clang__)
+#define META_IS_CONSTRUCTIBLE(...) __is_constructible(__VA_ARGS__)
+#elif defined(__GNUC__) && __GNUC__ >= 8
+#define META_IS_CONSTRUCTIBLE(...) __is_constructible(__VA_ARGS__)
```
Comment (CaseyCarter) [inline include/meta/meta_fwd.hpp]: This is identical to 186 - you could disjoin the conditions and eliminate the duplication. Also, MSVC supports this intrinsic.

File: include/range/v3/range/conversion.hpp [hunk 3]
```diff
@@ -206,8 +206,8 @@ namespace ranges
             template(typename Rng, typename Cont)
             concept convertible_to_container_container_impl_,
                 range<Cont> && (!view_<Cont>) && move_constructible<Cont> &&
-                is_true<ranges::defer::range<range_value_t<Cont>> &&
-                    !ranges::defer::view_<range_value_t<Cont>>> &&
+                bool(ranges::defer::range<range_value_t<Cont>> &&
+                    !ranges::defer::view_<range_value_t<Cont>>) &&
```
Comment (CaseyCarter) [inline include/range/v3/range/conversion.hpp]: `bool(...)` is not a primary expression - this needs parentheses.

File: include/meta/meta_fwd.hpp (hunk 1 above)
Comment (ericniebler) [reply include/meta/meta_fwd.hpp]: Clang defines `__GNUC__`. I didn't know about MSVC, thanks.

File: include/range/v3/range/conversion.hpp (hunk 3 above)
Comment (ericniebler) [reply include/range/v3/range/conversion.hpp]: Bah.

File: include/meta/meta_fwd.hpp (hunk 1 above)
Comment (CaseyCarter) [reply include/meta/meta_fwd.hpp]: > Clang defines `__GNUC__`

Only in "pretending to be GCC" mode. In "pretending to be MSVC" mode it defines `_MSC_VER` instead. But yes, I think `#if defined(__GNUC__) || defined(__MSC_VER__)` covers all three.

Comment (ericniebler) [reply include/meta/meta_fwd.hpp]: I didn't know that!

**Comment by CaseyCarter** [comment]
FYI, I fixed the merge conflict with #1263.

**Pull request merged**